src/app/components/authmodel/AuthModel.js

```diff
@@ -0,0 +1,226 @@
+/* eslint-disable react/prop-types */
+'use client'
+
+import { useState, useEffect, React } from 'react'
+import { signIn, signUp,  confirmSignUp} from 'aws-amplify/auth'
+
+const AuthModel = ({ showModal, setShowModal, authStates, setAuthStates, setIsSignin}) => {
+
+  if (!showModal) return null
+
+  const [username, setUsername] = useState('')
+  const obj = {
```

Comment:
`obj` is not a intuitive name
```suggestion
  const authForms = {
```